jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/WebAppContext.java

```diff
@@ -1415,6 +1453,6 @@ private static void addClasses(ClassMatcher matcher, String attribute, Server se
         int l = classes.length;
         classes = Arrays.copyOf(classes, l + pattern.length);
         System.arraycopy(pattern, 0, classes, l, pattern.length);
-        server.setAttribute(attribute, classes);
+        ServletContextHandler.ENVIRONMENT.setAttribute(attributeKey, classes);
```

Comment:
Why are we setting this on an ENVIRONMENT?    Currently we should have system/server classes that can be set for:
 + all servers
 + a specific server
 + specific context

Perhaps this should be:
 + all servers
 + a specific server
 + a specific Environment
 + specific context

But this change looks to be just
 + all servers
 + a specific Environment
 + specific context